tst/EnergyPlus/unit/Psychrometrics.unit.cc

```diff
@@ -45,15 +45,150 @@
+
+    // Test 6: Label30 - TEMP. IS FROM   -60 C  TO   -40 C
+    H = -4.23e4 - 1.78637e4;
+    result = PsyTsatFnHPb_raw(H, PB);
+    actual_result = F6((H + 1.78637e4), -19.44, 8.53675e-4, -5.12637e-9, -9.85546e-14, -1.00102e-18, -4.2705e-24);
+    EXPECT_DOUBLE_EQ(actual_result, result);
+
+    // Test 7: TEMP. IS <  -60 C
+    H = -5.25e4 - 1.78637e4;
+    result = PsyTsatFnHPb_raw(H, PB);
+    actual_result = F6(-4.24e4, -19.44, 8.53675e-4, -5.12637e-9, -9.85546e-14, -1.00102e-18, -4.2705e-24);
+    EXPECT_DOUBLE_EQ(actual_result, result);
+
+    // Test 8: Label80 - TEMP. IS FROM   40 C  TO   60 C
+    H = 1.8380e5 - 1.78637e4;
+    result = PsyTsatFnHPb_raw(H, PB);
+    actual_result = F6((H + 1.78637e4), -1.29419, 3.88538e-4, -1.30237e-9, 2.78254e-15, -3.27225e-21, 1.60969e-27);
+    EXPECT_DOUBLE_EQ(actual_result, result);
+
+    // Test 9: Label90 - TEMP. IS FROM   60 C  TO   80 C
+    H = 4.7578e5 - 1.78637e4;
+    result = PsyTsatFnHPb_raw(H, PB);
+    actual_result = F6((H + 1.78637e4), 2.39214e1, 1.27519e-4, -1.52089e-10, 1.1043e-16, -4.33919e-23, 7.05296e-30);
+    EXPECT_DOUBLE_EQ(actual_result, result);
+
+    // Test 10: Label100 - TEMP. IS FROM   80 C  TO   90 C
+    H = 1.5446e6 - 1.78637e4;
+    result = PsyTsatFnHPb_raw(H, PB);
+    actual_result = F6((H + 1.78637e4), 4.88446e1, 3.85534e-5, -1.78805e-11, 4.87224e-18, -7.15283e-25, 4.36246e-32);
+    EXPECT_DOUBLE_EQ(actual_result, result);
+
+    // Test 11: Label110 - TEMP. IS FROM   90 C  TO   100 C
+    H = 3.8354e6 - 1.78637e4;
+    result = PsyTsatFnHPb_raw(H, PB);
+    actual_result = F7((H + 1.78637e4), 7.60565e11, 5.80534e4, -7.36433e-3, 5.11531e-10, -1.93619e-17, 3.70511e-25, -2.77313e-33);
+    EXPECT_DOUBLE_EQ(actual_result, result);
 
-namespace EnergyPlus {
+    // Test 12: TEMP > 100 C
+    H = 5.5867e7 - 1.78637e4;
+    result = PsyTsatFnHPb_raw(H, PB);
+    actual_result = F7(4.5866e7, 7.60565e11, 5.80534e4, -7.36433e-3, 5.11531e-10, -1.93619e-17, 3.70511e-25, -2.77313e-33);
+    EXPECT_DOUBLE_EQ(actual_result, result);
+
+    // Test 13: PB != 1.0133e5
+    H = 7.5223e4;
+    PB = 0.9133e5;
+    result = PsyTsatFnHPb_raw(H, PB);
+    actual_result = 23.445553;
+    EXPECT_NEAR(actual_result, result, 0.00001);
+
+    // Test 14: Cache version of the function - hit call
+    H = 7.5223e4 - 1.78637e4;
+    PB = 1.0133e5;
+    result = PsyTsatFnHPb_raw(H, PB);
+    Real64 cache_hit_result = PsyTsatFnHPb(H, PB);
+    EXPECT_DOUBLE_EQ(result, cache_hit_result);
+
+}
+
+TEST_F(EnergyPlusFixture, Psychrometrics_PsyTsatFnPb_Test)
+{
+
+    InitializePsychRoutines();
+
+    // Test 1: general
+    Real64 PB = 101325.0;
+    Real64 result = PsyTsatFnPb_raw(PB);
+    Real64 actual_result = 99.974;
+    EXPECT_NEAR(actual_result, result, 0.001);
+
+    // Test 2: Cache version of the function - first call
+    PB = 101325.0;
+    result = PsyTsatFnPb_raw(PB);
+    Real64 cache_result = PsyTsatFnPb(PB);
+    EXPECT_DOUBLE_EQ(result, cache_result);
```

Comment:
I do like this, however, where you simply compare the raw and cached versions.  It doesn't verify that the value is correct, but it does verify that the caching produces the same as the non-cached, which is very important here.